### PR TITLE
Add a Podman runner

### PR DESCRIPTION
**Pull request by agrare**
Use `podman secret create NAME -` to securely pass secrets to a pod

### CODE REVIEW

File: lib/manageiq/floe/workflow/runner/podman.rb [hunk 1]
```diff
@@ -3,6 +3,32 @@ module Floe
...
         class Podman < ManageIQ::Floe::Workflow::Runner
+          def run!(resource, env = {}, secrets = {})
+            raise ArgumentError, "Invalid resource" unless resource&.start_with?("docker://")
+
+            image = resource.gsub("docker://", "")
+
+            secret_guid = nil
+
+            if secrets && !secrets.empty?
+              require "securerandom"
```
Comment (Fryguy) [inline lib/manageiq/floe/workflow/runner/podman.rb:14]: Should probably move this to initialize or something (or just require all the time), so it's not in the hot path.

File: lib/manageiq/floe/workflow/runner/podman.rb [hunk 2]
```diff
@@ -3,6 +3,32 @@ module Floe
...
+              secret_guid = SecureRandom.uuid
+              AwesomeSpawn.run!("podman", :params => ["secret", "create", secret_guid, "-"], :in_data => secrets.to_json)
```
Comment (Fryguy) [inline lib/manageiq/floe/workflow/runner/podman.rb:16]:
```suggestion
            params = ["secret", "create", secret_guid, "-"]
            logger.debug("Running podman: #{AwesomeSpawn.build_command_line("podman", params)}")
            AwesomeSpawn.run!("podman", :params => params, :in_data => secrets.to_json)
```

Now that I wrote that I feel like this is a common logging situation - I'm wondering if AwesomeSpawn itself should have a :log_execution parameter or something.

File: lib/manageiq/floe/workflow/runner/podman.rb [hunk 3]
```diff
@@ -3,6 +3,32 @@ module Floe
...
+            end
+
+            params = ["run", :rm]
+            params += env.map { |k, v| [:e, "#{k}=#{v}"] } if env && !env.empty?
+            params << [:secret, secret_guid] if secret_guid
+            params << image
+
+            require "awesome_spawn"
+            logger.debug("Running podman: #{AwesomeSpawn.build_command_line("docker", params)}")
```
Comment (Fryguy) [inline lib/manageiq/floe/workflow/runner/podman.rb:25]:
```suggestion
            logger.debug("Running podman: #{AwesomeSpawn.build_command_line("podman", params)}")
```

File: spec/workflow/runner/podman_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    it "passes a secrets volume to podman run" do
+      expect(AwesomeSpawn).to receive(:run!).with("podman", :params => array_including("secret", "create", "-"), :in_data => {"luggage_password" => "1234"}.to_json)
+      expect(AwesomeSpawn)
+        .to receive(:run!)
+        .with("podman", :params => array_including("run", :rm, [:e, "FOO=BAR"], [:secret, anything], "hello-world:latest"))
+        .and_return(result)
+      expect(AwesomeSpawn).to receive(:run).with(a_string_including("podman secret rm")).and_return(result)
+
+      subject.run!("docker://hello-world:latest", {"FOO" => "BAR"}, {"luggage_password" => "1234"})
```
Comment (Fryguy) [inline spec/workflow/runner/podman_spec.rb:41]: 😉 

```suggestion
      subject.run!("docker://hello-world:latest", {"FOO" => "BAR"}, {"luggage_password" => "12345"})
```

File: lib/manageiq/floe/workflow/runner/podman.rb (hunk 1 above)
Comment (agrare) [reply lib/manageiq/floe/workflow/runner/podman.rb:14]: Done in 06e64f8

**Pull request merged**